### PR TITLE
Create first-odata-api.md

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Internal review URL](https://review.docs.microsoft.com/en-us/odata/webapi/first-odata-api?branch=pr-en-us-83)
Fixes https://github.com/aspnet/AspNetCore.Docs/issues/15947

### CODE REVIEW

File: Odata-docs/TOC.yml [hunk 1]
```diff
@@ -167,6 +167,8 @@
     items:
     - name: Getting started
       href: /odata/webapi/netcore
+    - name: Build web APIs with OData
```
Comment (madansr7) [inline Odata-docs/TOC.yml]: How is this different from Getting Started document above? Can rename to a title like "Web  API with query options". Also  "with OData" seems unnecessary.

File: Odata-docs/webapi/first-odata-api.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+By [FIVIL](https://github.com/fivil) and [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+This sample:
+
+* Demonstrates how to add [OData](https://www.odata.org/) query options support in an ASP.NET Core Web API app.?
+* Uses the completed [to-do Web API](/aspnet/core/tutorials/first-web-api) as a starting point.
+* Does not use an [Entity Data Model](http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/part3-csdl/odata-v4.0-errata03-os-part3-csdl-complete.html#_Toc453752491) (EDM).
+
+A malicious or naive client may construct a query that consumes excessive resources. Such a query can disrupt access to your service. Review [Security Guidance for ASP.NET Core Web API OData](odata-security.md) before starting this tutorial.
```
Comment (madansr7) [inline Odata-docs/webapi/first-odata-api.md]: Can we change this to [!Note]. Seems important and should be emphasized.

File: Odata-docs/TOC.yml [hunk 3]
```diff
@@ -167,6 +167,8 @@
     items:
```
Comment (madansr7) [inline Odata-docs/TOC.yml]: Is there a reason the project files are checked in to the docs folders? Can we move them to GitHub?

File: Odata-docs/webapi/first-odata-api.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+ms.date: 05/13/2019
+---
+
+# Sample: Build web APIs with OData support using ASP.NET Core
+
+By [FIVIL](https://github.com/fivil) and [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+This sample:
+
+* Demonstrates how to add [OData](https://www.odata.org/) query options support in an ASP.NET Core Web API app.?
```
Comment (xuzhg) [inline Odata-docs/webapi/first-odata-api.md]: >app.? [](start = 107, length = 5)

finish the sentence

File: Odata-docs/TOC.yml (hunk 3 above)
Comment (madansr7) [inline Odata-docs/TOC.yml]: there are build errors and warnings that need to be resolved. can you please look into it.

File: Odata-docs/webapi/first-odata-api.md (hunk 4 above)
Comment (madansr7) [inline Odata-docs/webapi/first-odata-api.md]: There is a question mark at the end.

File: Odata-docs/TOC.yml (hunk 3 above)
Comment (Rick-Anderson) [reply Odata-docs/TOC.yml]: > there are build errors and warnings that need to be resolved. can you please look into it.

[Link to build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C17%5C2ea817ce-bd30-41f2-f2eb-9eb93b4d61ef%5CPullRequest%5C201912170118339753-83%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=rQ4Jyim8JUiGXuUB9Sk5IJlgcKIPYEn55%2FGE4dcoJ%2Fk%3D&st=2019-12-17T01%3A14%3A45Z&se=2020-01-17T01%3A19%3A45Z&sp=r)

There are no build `Warning` or `Suggestion` from the files in this PR. I fixed all the build errors in your repro in my previous PR. Most of the warnings are `Missing attribute: author. Add the current author's GitHub ID.`

Comment (Rick-Anderson) [reply Odata-docs/TOC.yml]: > Is there a reason the project files are checked in to the docs folders? Can we move them to GitHub?

This is GitHub. They need to be in this repo to import as snippets.  For example, the following line
` [!code-csharp[](first-odata-api/samples/2.2/TodoApi/Startup.cs?highlight=6-7&name=snippet_dic)]`

That's how we do all our docs at https://github.com/aspnet/AspNetCore.Docs

See my [ODATA Expand](https://github.com/MicrosoftDocs/OData-docs/pull/74/files) PR you merged.

**Comment by Rick-Anderson** [comment]
@madansr7 what do we need to get this published?  @figuerres  at https://github.com/aspnet/AspNetCore.Docs/issues/16704 is asking.

**Comment by Rick-Anderson** [comment]
@madansr7 what do we need to get this published? @figuerres at aspnet/AspNetCore.Docs#16704 is asking.

**Pull request merged**